### ISSUE TITLE
Stream interfaces checking if execution_space is valid

### ISSUE THREAD

**Issue by vqd8a** [kokkos/kokkos-kernels#1648]
I am just looking at the stream interface in ```swap```:
https://github.com/kokkos/kokkos-kernels/blob/4c06b6cbd14114aee9db80de84d32019236e4d7d/blas/src/KokkosBlas1_swap.hpp#L70
Should we check if the ```execution_space``` is valid first?

**Comment by vqd8a** [comment]
@ndellingwood Yes, it is. I created this issue a while ago just to remind @lucbv of adding checking for valid ```execution_space``` in other BLAS stream interfaces too.

**Comment by ndellingwood** [comment]
Thanks @vqd8a ! I'll label as enhancement, please relabel if you prefer another option
